rest/amino.go

```diff
@@ -0,0 +1,111 @@
+package rest
+
+import (
+	"fmt"
+	"net/url"
+
+	clienttx "github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/gin-gonic/gin"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+)
+
+func getPage(query url.Values) (uint64, error) {
+	page, err := getUint(query, "page")
+	if page == 0 {
+		page = 1
+	}
+	return page, err
+}
+
+func convertToStdTx(txBytes []byte) (legacytx.StdTx, error) {
+	txI, err := encodingConfig.TxConfig.TxDecoder()(txBytes)
+	if err != nil {
+		return legacytx.StdTx{}, err
+	}
+
+	tx, ok := txI.(signing.Tx)
+	if !ok {
+		return legacytx.StdTx{}, fmt.Errorf("%+v is not backwards compatible with %T", tx, legacytx.StdTx{})
+	}
+
+	return clienttx.ConvertTxToStdTx(encodingConfig.Amino, tx)
+}
+
+func packStdTxResponse(txRes *types.TxResponse) error {
+	txBytes := txRes.Tx.Value
+	stdTx, err := convertToStdTx(txBytes)
+	if err != nil {
+		return err
+	}
+
+	// Pack the amino stdTx into the TxResponse's Any.
+	txRes.Tx = codectypes.UnsafePackAny(stdTx)
+	return nil
+}
+
+func handleAminoTxsSearch(c *gin.Context, pool *pgxpool.Pool) {
+	q := c.Request.URL.Query()
+	page, err := getPage(q)
+	if err != nil {
+		c.AbortWithStatusJSON(400, gin.H{"error": err.Error()})
+		return
+	}
+	limit, err := getLimit(q, "limit")
+	if err != nil {
+		c.AbortWithStatusJSON(400, gin.H{"error": err.Error()})
+		return
+	}
+	events, err := getEvents(q)
+	if err != nil {
+		c.AbortWithStatusJSON(400, gin.H{"error": err.Error()})
+		return
+	}
+	conn, err := db.AcquireFromPool(pool)
+	if err != nil {
+		logger.L.Panicw("Cannot acquire connection from database connection pool", "error", err)
```

Comment:
🤔 I think here we should log with `Errorw` instead of `Panicw`, and then abort with 5xx